### PR TITLE
LOG-371. Bump Elasticsearch CRD to v1

### PR DESCRIPTION
**Pull request by jcantrill**
This PR bumps the Elasticsearch CRD for GA to resolve: https://jira.coreos.com/browse/LOG-371

### CODE REVIEW

File: deploy/openshift/admin-elasticsearch-template.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ objects:
       plural: elasticsearches
       singular: elasticsearch
     scope: Namespaced
-    version: v1alpha1
+    version: v1
```
Comment (ewolinetz) [inline deploy/openshift/admin-elasticsearch-template.yaml]: side note, not sure why we still have these files in the repo... todo in future?

File: pkg/apis/elasticsearch/v1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -2,12 +2,20 @@
...
 
-package v1alpha1
+package v1
 
 import (
 	runtime "k8s.io/apimachinery/pkg/runtime"
 )
 
+// DeepCopyObject is an autogenerated deepcopy function, copying the receiver, creating a new runtime.Object.
+func (in *ElasticsearchList) DeepCopyObject() runtime.Object {
```
Comment (ewolinetz) [inline pkg/apis/elasticsearch/v1/zz_generated.deepcopy.go]: nit, can you move this down to lines 81 - 88 where it was before?

**Comment by ewolinetz** [comment]
couple formatting nits, otherwise lgtm

**Comment by ewolinetz** [comment]
/test e2e-aws
terraform flake

File: deploy/openshift/admin-elasticsearch-template.yaml (hunk 1 above)
Comment (jcantrill) [reply deploy/openshift/admin-elasticsearch-template.yaml]: yes... there are a bunch of these I'm not sure if/where they are used.  they all really need to be condensed and consolidated to only use whatever ends up in the manifest

File: pkg/apis/elasticsearch/v1/types.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
-package v1alpha1
+package v1
 
 import (
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (richm) [inline pkg/apis/elasticsearch/v1/types.go]: is it ok to have an import for `v1` in an unrelated package `v1`?  CLO uses `corev1` in some places for this import

Comment (jcantrill) [reply pkg/apis/elasticsearch/v1/types.go]: as long as there is no conflict it doesnt matter.  compilation should catch any issues

**Pull request merged**